apps/heureka/src/hooks/useUrlState.js

```diff
@@ -21,19 +21,34 @@ const useUrlState = () => {
   const loggedIn = useAuthLoggedIn()
   const { setFiltersFromURL, syncFiltersWithURL } = useFilterActions()
 
-  const activeView = useGlobalsActiveNavEntry()
-  const { setShowPanel, setActiveView } = useGlobalsActions()
+  const activeView = useGlobalsActiveView()
   const detailsFor = useGlobalsShowPanel()
+  const { setShowPanel, setActiveView, setServiceDetail, setIssueDetail, syncDetailsWithURL } = useGlobalsActions()
 
   // Set initial state from URL (on login)
   useLayoutEffect(() => {
     if (isURLRead) return
 
     const urlState = urlStateManager.currentState()
     if (urlState) {
+      // Set the active view
+      if (urlState[constants.ACTIVE_VIEW]) {
+        setActiveView(urlState[constants.ACTIVE_VIEW])
+      } else {
+        setActiveView("Services") // Default active view
+      }
+
       setFiltersFromURL(urlState[constants.ACTIVE_FILTERS], urlState[constants.SEARCH_TERM])
-      if (urlState[constants.DETAILS_FOR]) setShowPanel(urlState[constants.DETAILS_FOR])
-      if (urlState[constants.ACTIVE_NAV]) setActiveView(urlState[constants.ACTIVE_NAV])
+
+      // Set panel and details for service/issue
+      if (urlState[constants.DETAILS_FOR]) {
```

Comment:
This section is a bit confusing and prone to errors. How about using a switch statement instead?

```
if (detailsFor) {
  setShowPanel(detailsFor);

  switch (detailsFor) {
    case constants.PANEL_SERVICE:
      setServiceDetail(urlState[constants.SERVICE_NAME]); // Show service details panel
      break;
    case constants.PANEL_ISSUE:
      setIssueDetail(urlState[constants.ISSUE_ID]); // Show issue details panel
      break;
  }
}
```